test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,243 @@
+}
+
+describe('HyperliquidInterchainToken', () => {
+    let hyperliquidInterchainToken, hyperliquidInterchainTokenDeployer;
+
+    const name = 'HyperliquidToken';
+    const symbol = 'HLT';
+    const decimals = 18;
+    const mintAmount = 123;
+
+    let token;
+    let owner;
+    let user;
+    let provider;
+    let tokenAddress;
+    let slot0;
+    let deployerFromSlot0;
+    let deployerFromContract;
+
+    before(async () => {
+        const wallets = await ethers.getSigners();
+        owner = wallets[0];
+        user = wallets[1];
+
+        hyperliquidInterchainToken = await deployContract(owner, 'HyperliquidInterchainToken', [owner.address]);
+        hyperliquidInterchainTokenDeployer = await deployContract(owner, 'InterchainTokenDeployer', [hyperliquidInterchainToken.address]);
+
+        const salt = getRandomBytes32();
+        const tokenId = getRandomBytes32();
+
+        tokenAddress = await hyperliquidInterchainTokenDeployer.deployedAddress(salt);
+
+        token = await getContractAt('HyperliquidInterchainToken', tokenAddress, owner);
+
+        await hyperliquidInterchainTokenDeployer
+            .deployInterchainToken(salt, tokenId, owner.address, name, symbol, decimals)
+            .then((tx) => tx.wait());
+
+        await token.mint(owner.address, mintAmount).then((tx) => tx.wait());
+        expect(await token.interchainTokenId()).to.equal(tokenId);
+
+        provider = ethers.provider;
+        slot0 = await provider.getStorageAt(tokenAddress, 0);
+        deployerFromSlot0 = bytes32ToAddress(slot0);
+        deployerFromContract = await token.deployer();
+    });
+
+    describe('Hyperliquid Interchain Token', () => {
+        it('Should calculate hardcoded constants correctly', async () => {
+            await expect(deployContract(owner, 'HyperliquidInterchainToken', [owner.address])).to.not.be.reverted;
+        });
+
+        it('should verify initial deployer state and slot 0 storage', async () => {
+            expect(deployerFromSlot0).to.equal(AddressZero);
+            expect(deployerFromContract).to.equal(AddressZero);
+            expect(slot0).to.equal('0x0000000000000000000000000000000000000000000000000000000000000000');
+
+            const currentDeployer = await token.deployer();
+            expect(deployerFromSlot0).to.equal(currentDeployer);
+            expect(deployerFromSlot0.toLowerCase()).to.equal(deployerFromContract.toLowerCase());
+        });
+
+        it('should update deployer and verify storage changes', async () => {
+            const initialSlot0 = slot0;
+            const newDeployer = user.address;
+
+            await token.connect(owner).updateDeployer(newDeployer);
+
+            const updatedDeployer = await token.deployer();
+            expect(updatedDeployer).to.equal(newDeployer);
+
+            const updatedSlot0 = await provider.getStorageAt(tokenAddress, 0);
+            expect(updatedSlot0).to.not.equal(initialSlot0);
```

Comment:
```suggestion
```
not needed, we're already checking that it equals the new deployer, which implies it's not the same